fvwm/expand.c

```diff
@@ -391,6 +394,34 @@ static signed int expand_vars_extended(
 			return -1;
 		}
 		if (*(rest + n) != 0)
+		{
+			/* Check for .lightenN or .darkenN */
+			csadj = 101;
+			l = -1;
+			if (sscanf(rest + n, ".lighten%d%n", &l, &nn) && l >= 0 && l <= 100)
+			{
+				csadj = l;
+			}
+			if (sscanf(rest + n, ".darken%d%n", &l, &nn) && l >= 0 && l <= 100)
+			{
+				csadj = -l;
+			}
+			/* Check for .hash */
+			if (strcmp(rest + n + nn,  ".hash") == 0)
+			{
+				use_hash = True;
+				nn = nn + 5;
```

Comment:
Try and avoid any magic variables:

```suggestion
				nn += strlen(".hash");
```